packages/@aws-cdk/aws-stepfunctions/lib/state-machine.ts

```diff
@@ -98,8 +130,11 @@ export class StateMachine extends StateMachineBase {
         const graph = new StateGraph(props.definition.startState, `State Machine ${id} definition`);
         graph.timeout = props.timeout;
 
+        this.stateMachineType = props.stateMachineType ? props.stateMachineType : StateMachineType.STANDARD;
```

Comment:
Is the default in CloudFormation perhaps `STANDARD`? I think so, no? If so, I'd rather we default to `undefined` here, so that the churn on the existing CloudFormation templates is minimal.

The `@default` annotation on the `stateMachineType` property should STILL indicate that the default behavior is `STANDARD`. `@default` annotations concern themselves with behavior of the stack as a whole, not with the details of what is rendered to a CFN temlate.

Reply:
Thank you for your feedback.

In my latest commit, the member variable `stateMachineType` is set to `STANDARD` by default (line 133), while the parameter `stateMachineType` in the cloudformation resource keeps `undefined` if users do not specify this optional field.

This solution helps avoid breaking the existing integration tests. :)